trading_cost.py

```diff
@@ -0,0 +1,67 @@
+"""
+A file to build cost objects.
+"""
+
+class CurrencyCost:
+    """
+    Cost function of currency trading.
+    """
+    def __init__(self, base_ins, instruments):
+        self.base_ins = base_ins
+        self.cost = 0
+        self.profit = 0
+        self.quantity = {
+            instrument : 0 for instrument in instruments}
+        self.value = {
+            instrument : 0 for instrument in instruments}
+        self.value[base_ins] = 1
+        self.quantity[base_ins] = 1
+
+    def __call__(self, portfolio, price, liquidation):
+        self.reset()
+        if liquidation:
+            for instrument in portfolio[-1]:
+                portfolio[-1][instrument] = (
+                    1 if instrument is self.base_ins else 0)
+        self.calculate_profit(portfolio[-1], price)
+        return self.profit - self.cost, self.cost
+
+    def calculate_profit(self, today, price):
```

Comment:
Is it possible to let this function return `pnl, cost` direclty? If so, we don't need `self.profit`, `self.cost` and `self.reset` at all.
pnl (profit and loss) is the term people use to call `self.profit` because it is not always "profit".